L1Trigger/Phase2L1ParticleFlow/src/COEFile.cc

```diff
@@ -0,0 +1,127 @@
+#include "L1Trigger/Phase2L1ParticleFlow/interface/COEFile.h"
+
+using namespace l1tpf_impl;
+
+COEFile::COEFile(const edm::ParameterSet& iConfig)
+    : file(nullptr),
+      coeFileName(iConfig.getUntrackedParameter<std::string>("coeFileName", "")),
+      bset_string_(""),
+      ntracksmax(iConfig.getUntrackedParameter<unsigned int>("ntracksmax")),
+      phiSlices(iConfig.getParameter<std::vector<edm::ParameterSet>>("regions")[0].getParameter<uint32_t>("phiSlices")),
+      debug_(iConfig.getUntrackedParameter<int>("debug", 0)) {
+  file = fopen(coeFileName.c_str(), "w");
+  writeHeaderToFile();
+  bset_.resize(tracksize);
+}
+
+void COEFile::writeHeaderToFile() {
+  char depth_width[256];
+  sprintf(depth_width,
```

Comment:
please use `snprintf` rather than `sprintf` to protect against buffer overflows

Reply:
ok